### PR TITLE
[WBCAMS-287] elastic job searches must have a salary

### PR DESCRIPTION
**Pull request by jonathan-longe**
This pull request changes the default Elastic Search query so that jobs must have a salary and removes from the GUI the ability to search for jobs without a salary.  I'm taking a light touch to the changes to ensure this change passes QA as quickly as possible as I expect the client will want to push this into PROD as quickly as possible.

### CODE REVIEW

**Comment by jonathan-longe** [comment]
@infojunkie as discussed this morning, waiting for approval from the client before merging

**Review by infojunkie — APPROVED** [review]
Lgtm, thanks

**Pull request merged**
